### PR TITLE
DOC: update naming of packages

### PR DESCRIPTION
**Pull request by FRidh**
###### Motivation for this change
- Clarify what the name of a package is when the package when the package is not a release but a commit from a repository (6fc624b, #12539)
- Drop the lowercase requirement and instead do stick to upstream name since this should minimize confusion and make it easier when generating package sets (6fc624b, #11052).

### CODE REVIEW

**Comment by edolstra** [comment]
Hm, I'm not sure about dropping the lowercase rule. For predictability, it's nice to know that a package is called `firefox` and not `Firefox` (so doing `nix-env -i firefox` will just work).

**Comment by edolstra** [comment]
The Fedora guidelines seem reasonable (https://fedoraproject.org/wiki/Packaging:Naming?rd=Packaging:NamingGuidelines):

> Package names should be in lower case and use dashes in preference to underscores. You can take some cues from the name of the upstream tarball, the project name from which this software came, and the name which has been used for this package by other distributions/packagers in the past. You can also request guidance from the upstream developers. Do not just blindly follow those examples, however, as package names should strive to be consistent within Fedora more than consistent between distributions. 

But also:

> In Fedora packaging, the maintainer should use his/her best judgement when considering how to name the package. While case sensitivity is not a mandatory requirement, case should only be used where necessary. Keep in mind to respect the wishes of the upstream maintainers. If they refer to their application as "ORBit", you should use "ORBit" as the package name, and not "orbit". However, if they do not express any preference of case, you should default to lowercase naming.

File: doc/coding-conventions.xml [hunk 1]
```diff
@@ -244,23 +244,15 @@ bound to the variable name <varname>e2fsprogs</varname> in
   <listitem><para>Generally, try to stick to the upstream package
   name.</para></listitem>
 
-  <listitem><para>Don’t use uppercase letters in the
```
Comment (jagajaga) [inline doc/coding-conventions.xml]: I'm against that removing.

File: doc/coding-conventions.xml [hunk 2]
```diff
@@ -244,23 +244,15 @@ bound to the variable name <varname>e2fsprogs</varname> in
...
+  dash) — e.g., <literal>"hello-0.3.1rc2"</literal>.</para></listitem>
 
-  <listitem><para>If package is fetched from git's commit then
+  <listitem><para>If a package is not a release but a commit from a repository, then
   the version part of the name <emphasis>must</emphasis> be the date of that 
   (fetched) commit. The date must be in <literal>"YYYY-MM-DD"</literal> format.
-  Also add <literal>"git"</literal> to the name - e.g.,
-  <literal>"pkgname-git-2014-09-23"</literal>.</para></listitem>
+  Also append <literal>"unstable"</literal> to the name - e.g.,
+  <literal>"pkgname-unstable-2014-09-23"</literal>.</para></listitem>
```
Comment (jagajaga) [inline doc/coding-conventions.xml]: :+1:

**Comment by joachifm** [comment]
Also closes https://github.com/NixOS/nixpkgs/issues/14572

**Comment by FRidh** [comment]
Thanks for the comments. I pushed the change regarding `unstable` suffix to master. See https://github.com/NixOS/nixpkgs/issues/11052 for a general discussion of package naming.

**Pull request closed without merging**